internal/rootcoord/root_coord.go

```diff
@@ -2275,6 +2284,13 @@ func (c *Core) SegmentFlushCompleted(ctx context.Context, in *datapb.SegmentFlus
 			zap.Int64s("compactFrom", in.Segment.CompactionFrom), zap.Error(err))
 		return failStatus(commonpb.ErrorCode_UnexpectedError, err.Error()), nil
 	}
+	if in.Segment.CreatedByCompaction {
```

Comment:
what happens if we add lock but RemoveSegments err out?
Will we leave the lock there?